### DISCUSSION TITLE
why the component api  `/v1/component/project/{uuid}` responses just have 100 component ?

### DISCUSSION THREAD

**Discussion by gwarloki** [DependencyTrack/dependency-track#1851] [General]
swagger api about component  get `/v1/component/project/{uuid}` responses only 100 component。heard add `X-Total-Count` `1000` not  effective。
use `/v1/component/project/{uuid}?searchText=&pageSize=1000&pageNumber=1` responses 394 is ok.

**Comment by nscuro** [comment]
Almost all API endpoints use paging. The default page size is 100, and as you discovered with `/v1/component/project/{uuid}?searchText=&pageSize=1000&pageNumber=1`, the page parameters can be controlled using `pageSize` and `pageNumer`. You can use the same parameters for the `/v1/component/project/{uuid}` endpoint.

The Swagger documentation is autogenerated and wildly inaccurate in some areas. There's an open issue #840 to address these inaccuracies.

**Reply by jimklimov** [reply]
Also note the details in #209
